### PR TITLE
feat: Allow admins to filter bookings only by attendee emails

### LINKED ISSUE #13235: [CAL-2936] API: Allow admins to filter bookings only by attendee emails
We need to ignore the admin userId from the request when the `attendeeEmail` param is specified.
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2936](https://linear.app/calcom/issue/CAL-2936/api-allow-admins-to-filter-bookings-only-by-attendee-emails)</sub>

### PR DESCRIPTION
**Pull request by keithwillcode**
## What does this PR do?

Fixes #13235  

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Ensure an admin can get data back for attendees by email only

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/api/pages/api/bookings/_get.ts [hunk 1]
```diff
@@ -189,7 +189,7 @@ async function handler(req: NextApiRequest) {
       const userEmails = users.map((u) => u.email);
       args.where = buildWhereClause(userId, attendeeEmails, userIds, userEmails);
     } else if (filterByAttendeeEmails) {
-      args.where = buildWhereClause(userId, attendeeEmails, [], []);
+      args.where = buildWhereClause(null, attendeeEmails, [], []);
```
Comment (keithwillcode) [inline apps/api/pages/api/bookings/_get.ts:192]: This was always passing the admin's user ID meaning the results returned only included bookings where the admin was included in the booking.

File: apps/api/pages/api/bookings/_get.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ function buildWhereClause(
   userEmails: string[] = []
 ) {
   const filterByAttendeeEmails = attendeeEmails.length > 0;
-  const userFilter = userIds.length > 0 ? { userId: { in: userIds } } : { userId };
+  const userFilter = userIds.length > 0 ? { userId: { in: userIds } } : !!userId ? { userId } : {};
```
Comment (keithwillcode) [inline apps/api/pages/api/bookings/_get.ts:128]: This formatting is horrendous. Will validate functionality and likely refactor.

**Review by exception — DISMISSED** [review]

File: apps/api/pages/api/bookings/_get.ts (hunk 2 above)
Comment (alishaz-polymath) [reply apps/api/pages/api/bookings/_get.ts:128]: Would need to update the `userId` type in the function to `number | null`

**Review by alishaz-polymath — APPROVED** [review]

**Comment by keithwillcode** [comment]
Important: Don't merge until after v3.7 is released today.

**Pull request merged**